src/kontrol/__main__.py

```diff
@@ -62,6 +64,14 @@ def main() -> None:
     args = parser.parse_args()
     logging.basicConfig(level=_loglevel(args), format=_LOG_FORMAT)
 
+    expected_k_version = KVersion.parse(f'v{pyk.K_VERSION}')
+    actual_k_version = k_version()
+
+    if expected_k_version != actual_k_version:
+        print(
+            f'WARNING: An outdated version of K (v{actual_k_version}) is being used. Updating K to v{expected_k_version} is RECOMMENDED'
+        )
```

Comment:
Please extract a function `_check_k_version`.

Reply:
From where?

Reply:
I mean you can put this code in a function definition:

```python3
def _check_k_version() -> None:
    expected_k_version = ...
    ...
```

and then call the function in `main`:

```python3
def main() -> None:
    ...
    _check_k_version()
```

This improves readability.

Reply:
Ahh ok got it! Thanks! I just added the function in `__main__.py` in the Kontrol repo, but maybe it would make sense to add this function in the pyk repo here: https://github.com/runtimeverification/pyk/blob/master/src/pyk/kbuild/utils.py
What do you think?